packages/flutter_tools/lib/src/commands/build_bundle.dart

```diff
@@ -37,11 +37,13 @@ class BuildBundleCommand extends BuildSubCommand {
         hide: !verboseHelp,
         help: 'Track widget creation locations. Requires Dart 2.0 functionality.',
       )
-      ..addFlag('build-snapshot',
+      ..addOption('precompile',
         hide: !verboseHelp,
-        defaultsTo: false,
-        help: 'Build and use application-specific VM snapshot instead of\n'
-            'prebuilt one provided by the engine.',
+        help: 'Precompile functions specified in input file. This flag requires\n'
```

Comment:
Does this flag have to be specified when using `--dynamic` option? Can it only be specified when using `--dynamic`? Can we add runtime checks to make sure that flutter fails unless right combination is specified?

Reply:
Clarified help text and added a check in code.